Apps/phradar/radar_connector.py

```diff
@@ -0,0 +1,451 @@
+        self.save_progress("Connect to Radar")
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        radar_val, response = self._make_rest_call("/incidents", action_result, method="get", params=None)
+        if phantom.is_fail(radar_val):
+            self.save_progress("Radar test connectivity fail.")
+            return action_result.get_status()
+
+        self.save_progress("Radar test connectivity success")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_create_privacy_incident(self, param):
+        self.save_progress(f"Run action {self.get_action_identifier()}")
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        name = param.get("name")
+        if not name:
+            self.debug_print(f"required name param not present in action: {self.get_action_identifier()}")
+            return action_result.set_status(phantom.APP_ERROR, "incident name not specified")
+
+        description = param.get("description", "Privacy incident created by Splunk Phantom")
+
+        # use timezone set in asset configuration to set discovered date_time timezone
+        config = self.get_config()
+        incident_group = int(config.get("radar_incident_group"))
+        time_zone = config.get("time_zone", "UTC")
+        discovered = datetime.now(pytz.timezone(time_zone)).strftime("%Y-%m-%dT%H:%M:%S")
+
+        # get incident channel information
+        phantom_base_url = self._get_system_settings()["company_info_settings"]["fqdn"]
+        if not phantom_base_url:
+            return action_result.set_status(phantom.APP_ERROR, "Base URL for phantom appliance must be configured")
+        container_path = ""
+        container_id = self.get_container_id()
+        if container_id is not None:
+            container_path = f"/mission/{container_id}"
+
+        uri = f"{phantom_base_url}{container_path}"
+
+        self.save_progress(f"Create payload")
+
+        body = {
+            "incident_group_id": incident_group,
+            "channel": {
+                "id": f"{container_id}",
+                "source": "Splunk Phantom",
+                "uri": uri
+            },
+            "name": name,
+            "description": description,
+            "discovered": {
+                "date_time": discovered,
+                "time_zone": time_zone
+            }
+        }
+
+        radar_val, data = self._make_rest_call("/incidents", action_result, method="post", data=json.dumps(body))
+
+        if phantom.is_fail(radar_val):
+            return action_result.get_status()
+
+        # construct incident output with fields we want to render
+        incident = dict()
+        incident["incident_id"] = data["id"]
```

Comment:
Could throw an exception and cause a failure. Use a .get() with a default value or add at try/catch block.